src/Standards/PSR12/Tests/ControlStructures/ControlStructureSpacingUnitTest.inc

```diff
@@ -98,3 +98,24 @@ $expr2 &&
   $expr3) {
       // structure body
 };
+
+// Ensure the sniff handles too many newlines (not just too few). This was copied from src/Standards/Squiz/Tests/ControlStructures/ForLoopDeclarationUnitTest.1.inc
+for (
+
+
+    $i = 0
+
+
+    ;
+
+
+    $i < 10
+
+
+    ;
+
+
+    $i++
+
+
+) {}
```

Comment:
The new lines between the condition parts are irrelevant for testing this sniff, so this test could be simplified a little:

```suggestion
for (


    $i = 0;
    $i < 10;
    $i++


) {}
```

And what about also including a test which has a comment above the first condition ?
AFAICS, this should be handled correctly, but would be good to safeguard that.

```php
for (


    // Comment.
    $i = 0;
    $i < 10;
    $i++
) {}
```